src/python/py/models/builder.py

```diff
@@ -2613,6 +2642,48 @@ def make_layer(self, layer_id, layer):
             self.layernorm_attrs["last_layernorm"] = True
 
 
+class ChatGLMModel(Model):
+    def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
+        super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
+        # self.input_shapes["position_ids"] = [1]  # Note: This is optional and only needed if you want position_ids to be an int instead of a 2D tensor
+        self.layernorm_attrs["simple"] = True # RMSNorm in ChatGLM
+        self.rotemb_attrs["num_heads"] = self.num_attn_heads
+        self.rotemb_attrs["partial_rotary_factor"] = 0.5 # Line 755 of modeling_chatglm.py check self.rotary_pos_emb declaration
+        self.rotemb_attrs["rotary_embedding_dim"] = int(self.head_size * self.rotemb_attrs["partial_rotary_factor"])
+        self.rotemb_attrs["interleaved"] = 1
+        self.mlp_attrs["use_proj"], self.mlp_attrs["use_fc"] = True, False
```

Comment:
The defaults are the same as below so I think we can remove this.

https://github.com/microsoft/onnxruntime-genai/blob/f76867300d35a9e3ee66a2779b965700ea256e73/src/python/py/models/builder.py#L250-L252